lib/broadway_rabbitmq/producer.ex

```diff
@@ -77,11 +83,40 @@ defmodule BroadwayRabbitmq.Producer do
   end
 
   @impl true
+  def terminate(_reason, state) do
+    %{client: client, channel: channel} = state
+
+    if channel do
+      client.close_connection(channel.conn)
+    end
+
+    :ok
+  end
+
+  @impl Acknowledger
   def ack(channel, successful, failed) do
     ack_messages(successful, channel, :ack)
     ack_messages(failed, channel, :reject)
   end
 
+  @impl Producer
+  def prepare_for_draining(%{channel: nil}) do
```

Comment:
We can the channel be nil when draining? When we are waiting on back-off waiting for connections?

Reply:
Yes.